src/main/java/uk/gov/migration/R__11_Migrate_current_keys_to_index.java

```diff
@@ -32,46 +32,56 @@ public void migrate(Connection connection) throws Exception {
 		dbi.registerContainerFactory(new OptionalContainerFactory());
 		Handle handle = dbi.open();
 		
-		DataAccessLayer dataAccessLayer = new PostgresDataAccessLayer(
+		PostgresDataAccessLayer dataAccessLayer = new PostgresDataAccessLayer(
 				handle.attach(EntryQueryDAO.class),
 				handle.attach(IndexDAO.class),
 				handle.attach(IndexQueryDAO.class),
-				null,
-				null,
-				null,
-				null,
-				connection.getSchema());
+				handle.attach(EntryDAO.class),
+				handle.attach(EntryItemDAO.class),
+				handle.attach(ItemQueryDAO.class),
+				handle.attach(ItemDAO.class),
+				connection.getSchema(),
+				new IndexDriver(),
+				ImmutableMap.of(EntryType.user, Arrays.asList(new RecordIndexFunction(IndexNames.RECORD))));
 
 		RegisterReadOnly register = new RecordIndexRegister(dataAccessLayer);
-		IndexDriver indexDriver = new IndexDriver(dataAccessLayer);
+		IndexDriver indexDriver = new IndexDriver();
+
+		int recordCount = register.getTotalDerivationRecords(IndexFunctionConfiguration.IndexNames.RECORD);
+		Map<String, Record> indexRecords = register.getDerivationRecords(recordCount, 0, IndexNames.RECORD).stream().collect(Collectors.toMap(k -> k.getEntry().getKey(), v -> v));
```

Comment:
Can we get this information from `dataAccessLayer`? Then we wouldn't need the `RecordIndexRegister` private class at all.